project/src/main/scala/locales/ScalaLocaleCodeGen.scala

```diff
@@ -336,6 +424,53 @@ object ScalaLocaleCodeGen {
 
     Some(CalendarPatterns(datePatterns.getOrElse(Nil), timePatterns.getOrElse(Nil)))
   }
+
+  // from="1940-05-10" to="1986-01-01"
+  private val sdf: SimpleDateFormat = {
+    val df = new SimpleDateFormat("yyyy-MM-dd")
+    df.setTimeZone(TimeZone.getTimeZone("GMT"))
+    df
+  }
+
+  def parseCurrencyData(xml: Node): CurrencyData = {
+    def toOptionInt(node: NodeSeq): Option[Int] =
+      if (node.nonEmpty && node.text.nonEmpty) Some(node.text.toInt) else None
+    def toOptionDate(node: NodeSeq): Option[Date] =
+      if (node.nonEmpty && node.text.nonEmpty) Some(sdf.parse(node.text)) else None
+    def toOptionBoolean(node: NodeSeq): Option[Boolean] =
+      if (node.nonEmpty && node.text.nonEmpty) Some(node.text.toBoolean) else None
+
+    val fractions = (for {
+      info <- xml \ "currencyData" \ "fractions" \\ "info"
+    } yield {
+      val currencyCode = (info \ "@iso4217").text
+      val digits = (info \ "@digits").text.toInt
+      val rounding = (info \ "@rounding").text.toInt
+      val cashDigits = toOptionInt(info \ "@cashDigits")
+      val cashRounding = toOptionInt(info \ "@cashRounding")
+      CurrencyDataFractionInfo(currencyCode, digits, rounding, cashDigits, cashRounding)
+    })
+
+    val regions = (for {
+      region <- xml \ "currencyData" \\ "region"
+      currency <- region \\ "currency"
+    } yield {
+      val countryCode = (region \ "@iso3166").text
+
+      val currencyCode = (currency \ "@iso4217").text
+      // TODO: why am I getting random compile errors on these two fields?
+      val from = None // toOptionDate(currency \ "@from")
```

Comment:
that was the original point of having the toOptionDate...to check to see if it existed or not...I was getting random parse errors because it was parsing some non-date data...I'm not sure if that was transient or not